OneCommunityGlobal/HighestGoodNetworkApp
# Tim bmdashboard entry point

# Description
Implement Phase 2 Entry Point: 
- URL to access the Building Management Dashboard
- Password protection to access the Dashboard

## Related PRS (if any):
To test this PR you need to checkout the [backend PR 543](https://github.com/OneCommunityGlobal/HGNRest/pull/543)

## Main changes explained:
- Adds URL entry point “/bmdashboard” to access Building Management Dashboard (ie Phase 2)
- Users must be logged in AND verified via login page at “bmdashboard/login” to access the BM Dashboard.

## How to test:
- Follow steps to test backend PR 543
- `git checkout Tim_bmdashboard_entry_point`
- `npm install`
- `npm run start:local`
- Log out of HGN and test that attempting to access “/bmdashboard” or “/bmdashboard/:whatever” redirects to “/login”
- Log into HGN as a volunteer user
- Test that attempting to access “/bmdashboard” or “/bmdashboard/:whatever” redirects to “bmdashboard/login”
- Check frontend validation of login form: 
   - Enter text that is not in correct email format (Note: a string with only two segments such as test@test is technically acceptable email format)
   - Enter an invalid password (< 8 characters)
- Check backend validation of login form: 
   - Enter an email for an account that does not exist, or is not the email you are signed in as
   - Enter the correct email but incorrect password
- Enter the correct email and password for the user you are already signed in as. Verify you are redirected to “/bmdashboard”.
- Also verify that attempting to access “bmdashboard/login” now redirects to “/bmdashboard”
- Finally, log out of HGN and verify you can no longer access “/bmdashboard”

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/90e6d0ca-88f8-4520-8d03-52bbf55ff1c1

## Note:
- There are no currently no role restrictions for access the BM Dashboard, so all testing can be done as a Volunteer user
